.php_cs.dist

```diff
@@ -81,23 +81,26 @@ $config = PhpCsFixer\Config::create()
 
 // special handling of fabbot.io service if it's using too old PHP CS Fixer version
 if (false !== getenv('FABBOT_IO')) {
-    try {
-        $fixerFactory = PhpCsFixer\FixerFactory::create()
-            ->registerBuiltInFixers();
-        $config->setRules(array_filter(
-            $config->getRules(),
-            function ($rule) use ($fixerFactory) {
-                return $fixerFactory->hasRule($rule);
-            },
-            ARRAY_FILTER_USE_KEY
-        ));
-    } catch (PhpCsFixer\ConfigurationException\InvalidConfigurationException $e) {
-        $config->setRules([]);
-    } catch (UnexpectedValueException $e) {
-        $config->setRules([]);
-    } catch (InvalidArgumentException $e) {
-        $config->setRules([]);
-    }
+    $fixerFactory = PhpCsFixer\FixerFactory::create();
+    $fixerFactory->registerBuiltInFixers();
+    $ruleSet = new PhpCsFixer\RuleSet($config->getRules());
+    $config->setRules(array_filter(
+        $ruleSet->getRules(),
+        function ($config, $name) use ($fixerFactory) {
+            try {
+                $fixerFactory->useRuleSet(new PhpCsFixer\Ruleset([$name => $config]));
```

Comment:
1. will crash
```diff
-Ruleset
+RuleSet
```

2. won't work as you think:
```php
<?php

$fixerFactory = PhpCsFixer\FixerFactory::create();
$fixerFactory->registerBuiltInFixers();

$fixerFactory->useRuleSet(new PhpCsFixer\RuleSet(['strict_param' => true]));
$fixerFactory->useRuleSet(new PhpCsFixer\RuleSet(['braces' => true]));

> In FixerFactory.php line 167:
>  Rule "braces" does not exist.  
```


this proves one thing - you need to move this logic from config file to internal class, and then add tests for it.